src/wp-includes/class-wp-block-patterns-registry.php

```diff
@@ -165,7 +165,11 @@ public function get_registered( $pattern_name ) {
 			return null;
 		}
 
-		return $this->registered_patterns[ $pattern_name ];
+		$pattern            = $this->registered_patterns[ $pattern_name ];
+		$blocks             = parse_blocks( $pattern['content'] );
+		$visitor            = _parsed_block_visitor( $pattern ); // TODO: Should we use different functions for template vs pattern?
```

Comment:
Probably something like

```suggestion
		$visitor            = _parsed_block_visitor_for_pattern( $pattern ); // TODO: Use nicer factory name!
```

In that function, apply a `insert_hooked_block_into_pattern` filter (akin to https://github.com/WordPress/wordpress-develop/pull/5158/files#r1325968103).